metadata-ingestion/src/datahub/ingestion/source_config/operation_config.py

```diff
@@ -84,3 +109,28 @@ def is_profiling_enabled(operation_config: OperationConfig) -> bool:
         )
         return False
     return True
+
+
+def is_weekly_stripping_enabled(operation_config: OperationConfig) -> bool:
+    if (
+        operation_config.weekly_striping_enabled
+        and operation_config.lower_freq_profile_enabled is True
+    ):
+        return True
+    return False
+
+
+def is_hash_mod_matching_weekday(datasetUrn: str) -> bool:
+
+    """
+    Checks if the consistent hash and mod 7 value equals weekday.
+    Args:
+        datasetUrn or table name
+    Returns:
+        bool: True if the hash value corresponds to the target weekday, False otherwise.
+    """
+
+    md5_hash = int(hashlib.md5(str(datasetUrn).encode()).hexdigest(), 16) % 7
+    if md5_hash == datetime.date.today().weekday():
```

Comment:
We should probably use utc here, or take in a configurable timezone.

Also, this can be replaced currently by `return md5_hash == datetime.date.today().weekday()`